test/e2e/main_test.go

```diff
@@ -65,6 +65,8 @@ type testContext struct {
 	providers.CloudProvider
 	// hasCustomVXLAN tells if the cluster is using a custom VXLAN port for communication
 	hasCustomVXLAN bool
+	// workloadNamespace is the namespace to deploy our test pods on
```

Comment:
why should this be a property of testContext struct?

Reply:
Because I asked for it. It was a global variable before this. But I am open to other suggestions.

Reply:
Ignore if that is the case, this is just a nit. I don't want to hold this PR just for that.